packages/bcp/src/cosmwasmconnection.ts

```diff
@@ -304,15 +283,37 @@ export class CosmWasmConnection implements BlockchainConnection {
     };
   }
 
-  public async searchTx(
-    query: TransactionQuery,
-  ): Promise<readonly (ConfirmedTransaction<UnsignedTransaction> | FailedTransaction)[]> {
-    const queryString = buildQueryString(query);
+  public async searchTx({
+    height,
+    id,
+    maxHeight,
+    minHeight,
+    sentFromOrTo,
+    signedBy,
+    tags,
+  }: TransactionQuery): Promise<readonly (ConfirmedTransaction<UnsignedTransaction> | FailedTransaction)[]> {
+    if ([maxHeight, minHeight, signedBy, tags].some(component => component !== undefined)) {
+      throw new Error("Transaction query by maxHeight, minHeight, signedBy or tags not yet supported");
+    }
+
+    if ([id, height, sentFromOrTo].filter(component => component !== undefined).length !== 1) {
+      throw new Error(
+        "Transaction query by id, height and sentFromOrTo is mutually exclusive. Exactly one must be set.",
```

Comment:
Height can be combined with others. Id and sentfromorto are indeed exclusive.

At least in wasmd node, I am not sure about in the cosmwasmclient

Reply:
CosmWasmClient defines distinct search cases (wich BCP should have as well) to better allow for proper implementation of specific use cases:

```ts
export interface SearchByIdQuery {
  readonly id: string;
}
export interface SearchByHeightQuery {
  readonly height: number;
}
export interface SearchBySentFromOrToQuery {
  readonly sentFromOrTo: string;
}
export declare type SearchTxQuery = SearchByIdQuery | SearchByHeightQuery | SearchBySentFromOrToQuery;
```

We can add the ability to combine height (not minHeight/maxHeight) with other queries but I don't see the use case for that right now.